ui/src/app/views/dataset/component.tsx

```diff
@@ -58,15 +92,102 @@ export const DataSet = withCurrentWorkspace()(class extends React.Component<
       const conceptSets = await Promise.resolve(conceptSetsApi()
         .getConceptSetsInWorkspace(namespace, id));
-      this.setState({conceptSetList: conceptSets.items});
+      this.setState({conceptSetList: conceptSets.items, loadingConceptSets: false});
     } catch (error) {
       console.log(error);
     }
   }
 
+  convertResource(r: ConceptSet | Cohort, rType: ResourceType): RecentResource {
+    const {workspace} = this.props;
+    this.setState({rType: rType});
+    console.log(rType);
+    return convertToResource(r, workspace.namespace, workspace.id,
+      workspace.accessLevel as unknown as WorkspaceAccessLevel, rType);
+  }
+
+  openConfirmDelete(r: ConceptSet, rType: ResourceType): void {
+    const rc = this.convertResource(r, rType);
+    this.setState({confirmDeleting: true, resource: rc});
+  }
+
+  closeConfirmDelete(): void {
+    this.setState({confirmDeleting: false});
+    if (this.state.rType === ResourceType.CONCEPT_SET) {
+      const deleted = this.state.resource.conceptSet;
+      const updatedList = fp.filter(conceptSet => conceptSet.id !== deleted.id,
+        this.state.conceptSetList);
+      this.setState({conceptSetList: updatedList});
+    }
+    this.setState({resource: undefined, rType: undefined});
+  }
+
+  edit(r: Cohort | ConceptSet, rType: ResourceType): void {
+    const rc = this.convertResource(r, rType);
+    console.log(rType);
+    this.setState({editing: true, resource: rc});
+  }
+
+  receiveDelete() {
+    const {resource, rType} = this.state;
+    let call;
+    const id = rType === ResourceType.COHORT ? resource.cohort.id : resource.conceptSet.id;
+    const argList = [resource.workspaceNamespace,
+      resource.workspaceFirecloudName, id];
+    if (rType === ResourceType.CONCEPT_SET) {
+      // these also throw errors in the logs
+      // functions correctly, and I think is so pretty, but
+      // I don't like all the errors in the logs, so torn.
+      call = conceptSetsApi().deleteConceptSet(...argList);
+    } else {
+      call = cohortsApi().deleteCohort(...argList);
+    }
+
+    call.then(() => this.closeConfirmDelete());
+  }
+
+  receiveEdit(resource: RecentResource): void {
+    const {rType} = this.state;
+    const updatedResource = rType === ResourceType.COHORT ? resource.cohort : resource.conceptSet;
+    const argList = [resource.workspaceNamespace,
+      resource.workspaceFirecloudName, updatedResource.id,
+      updatedResource];
+    let call;
+    if (resource.cohort) {
+      call = cohortsApi().updateCohort(...argList);
+    } else if (resource.conceptSet) {
+      this.setState({loadingConceptSets: true});
+      call = conceptSetsApi().updateConceptSet(...argList);
+    }
+    call.then(() => this.closeEditModal());
+  }
+
+  closeEditModal(): void {
+    const edited = this.state.resource.conceptSet;
+    const updatedList = this.state.conceptSetList.map((conceptSet) => {
+      return edited.id === conceptSet.id ? edited : conceptSet;
+    });
```

Comment:
Likewise to above, we should figure out whether we want to issue a reload or do the in-browser update. The only reason I bring it up is that I think we've done the reload in the past.

Reply:
And I think that's a bad reason to stick with something, I think moving forward we should use this, and have a tech debt ticket to change the other places.... Maybe should be a tech talk subject to get consensus from the team.

Reply:
Totally reasonable, I do think that this is probably the better way to do this, but it depends. If there’s a lot of people likely making edits on the same workspace/concept sets, then reloading is probably the way to do. A quick sync around it makes sense, but I’m comfortable going this way until we have a decision, and then adding a tech debt ticket to do cleanup one way or the other.